src/connection.js

```diff
@@ -1462,6 +1462,12 @@ export class Connection {
 
     url.protocol = url.protocol === 'https:' ? 'wss:' : 'ws:';
     url.host = '';
+    // Only shift the port by +1 as a convention for ws(s) only if given endpoint
+    // is explictly specifying the endpoint port (HTTP-based RPC), assuming
+    // we're directly trying to connect to solana-validator's ws listening port.
+    // When the endpoint omits the port, we're connecting to the protocol
+    // default ports: http(80) or https(443) and it's assumed we're behind a reverse
+    // proxy which manages WebSocket upgrade and backend port redirection.
```

Comment:
@mvines could you check my understanding?

Reply:
5b4aa213

Reply:
Yep, exactly right.  Thanks for adding this helpful detail

Reply:
> 5b4aa213

Menaa